runelite-client/src/main/java/net/runelite/client/plugins/poh/PohPlugin.java

```diff
@@ -147,4 +223,37 @@ public void onGameStateChanged(GameStateChanged event)
 			pohObjects.clear();
 		}
 	}
-}
+
+	@Subscribe
+	public void onAnimationChanged(AnimationChanged event)
+	{
+		Actor actor = event.getActor();
+		String actorName = actor.getName();
+		//Get actor who is doing light_burner anim
+		if (actor.getAnimation() == AnimationID.INCENSE_BURNER)
+		{
+			if (actor == client.getLocalPlayer())
+			{
+				int level = client.getRealSkillLevel(net.runelite.api.Skill.FIREMAKING);
+				countdownTimer = (200 + level) * ESTIMATED_TICK_LENGTH;
+				randomTimer = level * ESTIMATED_TICK_LENGTH;
+			}
+			else try
+			{
+				//On initial lookup client will freeze for ~1 tick to generate HiscoreKey,
+				//after that the hiscore lookup is mapped to a HiscoreKey (see HiscoreManager.java)
+				HiscoreResult playerStats = hiscoreManager.lookup(actorName, HiscoreEndpoint.NORMAL);
```

Comment:
you should probably not lookup playerStats in game thread

Reply:
which thread would you like me to utilize?

Reply:
Use ScheduledExecutorService#submit